pkg/datapath/linux/node.go

```diff
@@ -620,21 +620,31 @@ func (n *linuxNodeHandler) insertNeighbor(newNode *nodeTypes.Node, ifaceName str
 		}
 		if err := netlink.NeighSet(&neigh); err != nil {
 			scopedLog.WithError(err).Error("Failed to insert neighbor")
-			return
+			return err
 		}
 
+		scopedLog.Debug("Inserted new node neighbor")
+
 		n.neighByNextHop[nextHopStr] = &neigh
 		if option.Config.NodePortHairpin {
 			neighborsmap.NeighRetire(nextHopIPv4)
 		}
 	}
+
+	return nil
+}
+
+func (n *linuxNodeHandler) NodeNeighInsert(newNode nodeTypes.Node) error {
+	n.mutex.Lock()
+	defer n.mutex.Unlock()
+	return n.insertNeighbor(&newNode, n.getExternalFacingDeviceName())
 }
 
 // Must be called with linuxNodeHandler.mutex held.
-func (n *linuxNodeHandler) deleteNeighbor(oldNode *nodeTypes.Node) {
+func (n *linuxNodeHandler) deleteNeighbor(oldNode *nodeTypes.Node) error {
 	nextHopStr, found := n.neighNextHopByNode[oldNode.Identity()]
```

Comment:
@brb In an offline sync, you mentioned that we should allow the user to specify the node identity. Initially I assumed this meant the numeric identity, but the insert / delete neighbor code doesn't use the numeric identity. Instead it uses the `(*Node).Identity()` which is a combo of the node name and the cluster ID. 

Given this, I've made the API accept a node name. Is this what we want?

Reply:
For now it's OK to make the identity from the node name. We can always extend it if needed (by adding cluster name as a param).